### PR TITLE
KAFKA-17385: Remove authorizer, authorizer-properties, zk-tls-config-file in AclCommand

### PR DESCRIPTION
**Pull request by FrankYang0529**
The [authorizer](https://github.com/apache/kafka/blob/944c1353a925858ea9bd9024a713cd7301f55133/core/src/main/scala/kafka/admin/AclCommand.scala#L501-L506), [authorizer-properties](https://github.com/apache/kafka/blob/944c1353a925858ea9bd9024a713cd7301f55133/core/src/main/scala/kafka/admin/AclCommand.scala#L508-L514), and [zk-tls-config-file](https://github.com/apache/kafka/blob/944c1353a925858ea9bd9024a713cd7301f55133/core/src/main/scala/kafka/admin/AclCommand.scala#L610C63-L618) are deprecated options. We can consider to remove it in 4.0.

https://cwiki.apache.org/confluence/display/KAFKA/KIP-604%3A+Remove+ZooKeeper+Flags+from+the+Administrative+Tools

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 Could you please update `upgrade.html`?

**Comment by FrankYang0529** [comment]
> @FrankYang0529 Could you please update `upgrade.html`?

Added it. Thanks for the reminder.

**Comment by chia7712** [comment]
@FrankYang0529 please fix the conflicts

**Comment by FrankYang0529** [comment]
> @FrankYang0529 please fix the conflicts

Fixed it. Thank you.

File: core/src/main/scala/kafka/admin/AclCommand.scala
Comment (mimaison) [inline core/src/main/scala/kafka/admin/AclCommand.scala:197]: We can also remove the `AclCommandService` trait.

File: core/src/main/scala/kafka/admin/AclCommand.scala
Comment (mimaison) [inline core/src/main/scala/kafka/admin/AclCommand.scala:469]: Can we add a test for this?

**Review by mimaison — COMMENTED** [review]
Thanks for the PR. Can you also update security.html? See what I did in https://github.com/apache/kafka/pull/17347

**Comment by FrankYang0529** [comment]
Hi @mimaison, thanks for the review. I remove `AclCommandService` trait and update `security.html`.

File: core/src/main/scala/kafka/admin/AclCommand.scala
Comment (FrankYang0529) [reply core/src/main/scala/kafka/admin/AclCommand.scala:469]: I think `AclCommandTest#testUseWithoutBootstrapServerOptAndBootstrapControllerOpt` can cover this. Thank you.

File: core/src/test/java/kafka/admin/AclCommandTest.java
Comment (mimaison) [inline core/src/test/java/kafka/admin/AclCommandTest.java]: This field is now unused so can be deleted. Same for `ZOOKEEPER_CONNECT` below

**Review by mimaison — COMMENTED** [review]
Thanks for the updates. I left another comment

File: core/src/test/java/kafka/admin/AclCommandTest.java
Comment (FrankYang0529) [reply core/src/test/java/kafka/admin/AclCommandTest.java]: Removed them. Thanks for the review.

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

**Comment by mimaison** [comment]
I see https://issues.apache.org/jira/browse/KAFKA-17624 marked as in progress. Should we wait for it to be done before merging this?

**Comment by FrankYang0529** [comment]
> I see https://issues.apache.org/jira/browse/KAFKA-17624 marked as in progress. Should we wait for it to be done before merging this?

Yeah, I think we need to wait it to avoid related E2E cases fail.

cc @frankvicky

**Comment by frankvicky** [comment]
Hi @FrankYang0529 and @mimaison,

Thanks for the information. 
#17338 is now waiting for review. 
I believe it's almost done and just needs approval from the reviewer (@chia7712).

**Comment by chia7712** [comment]
@FrankYang0529 @mimaison @frankvicky Apologies for being particular, but I don't like to merge a PR that causes a known error. I'm going to merge this PR since the related E2E tests have already been removed.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**
